api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -250,7 +250,13 @@ message Bootstrap {
   // Setting this value causes failure if the
   // ``envoy.restart_features.use_apple_api_for_dns_lookups`` runtime value is true during
   // server startup. Apple' API only uses UDP for DNS resolution.
-  bool use_tcp_for_dns_lookups = 20;
+  bool use_tcp_for_dns_lookups = 20
+      [deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];
+
+  // Don't use the default search domains; only query hostnames as-is or as aliases for DNS lookups.
+  //  bool no_default_search_domain_for_dns_lookups = 30;
+
+  core.v3.AreaDnsLookupOptionFlags area_dns_lookup_option_flags = 30;
```

Comment:
For now I have kept the name AreaDnsLookupOptionFlags. I can change it to something shorter like DnsLookupOptions or DnsResolverOptions.

Reply:
Please add a comment for the new field.  You can add a `[TODO]` tag to remind yourself or others of future changes.  Grep for TODO throughout the codebase to see how users are tagged/self-tagged.